tests/subsys/net/lib/download_client/src/main.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <logging/log.h>
+LOG_MODULE_REGISTER(download_client_test, LOG_LEVEL_DBG);
+
+#include <zephyr.h>
+#include "net_private.h"
+#include <net/socket_offload.h>
+
+#include <ztest.h>
+#include <download_client.h>
+
+#include "mock/socket.h"
+#include "mock/dl_coap.h"
+
+static enum download_client_evt_id last_event = -1;
+
+static int download_client_callback(const struct download_client_evt *event)
+{
+	if (event == NULL) {
+		return -EINVAL;
+	}
+
+	switch (event->id) {
+	case DOWNLOAD_CLIENT_EVT_FRAGMENT:
+		last_event = DOWNLOAD_CLIENT_EVT_FRAGMENT;
+		break;
+	case DOWNLOAD_CLIENT_EVT_DONE:
+		last_event = DOWNLOAD_CLIENT_EVT_DONE;
+		break;
+	case DOWNLOAD_CLIENT_EVT_ERROR:
+		last_event = DOWNLOAD_CLIENT_EVT_ERROR;
+		break;
+	}
+
+	return 0;
+}
+
+static enum download_client_evt_id get_last_download_client_event(bool reset)
+{
+	enum download_client_evt_id event;
+
+	event = last_event;
+	if (reset) {
+		last_event = -1;
+	}
+
+	return event;
+}
+
+static void mock_return_values(const char *func, int32_t *val, size_t len)
+{
+	size_t i;
+
+	for (i = 0; i < len; i++) {
+		z_ztest_returns_value(func, val[i]);
+	}
+}
+
+static bool wait_for_event(enum download_client_evt_id event, uint8_t seconds)
+{
+	int i;
+
+	for (i = 0; i < (20 * seconds); i++) {
```

Comment:
Is the math correct here? 20 iterations with 10 ms wait time gives 200ms, not full second.